python/requirements/test-requirements.txt

```diff
@@ -71,7 +71,6 @@ werkzeug==2.1.2
 xlrd==2.0.1
 yq==3.2.2
 memray; platform_system != "Windows" and sys_platform != "darwin" and platform_machine != 'aarch64'
-memray @ git+https://github.com/bloomberg/memray.git; platform_system != "Windows" and sys_platform != "darwin" and platform_machine == 'aarch64'
```

Comment:
why is this line removed?

Reply:
memray cause a dependency conflicts in arm64; i just remove it since none of the arm jobs require it

Reply:
could you double check with places that uses it?

it seems that serve has a "memory profiling" feature, where it tries to import memray

@shrekris-anyscale @edoakes

Reply:
introduced here: https://github.com/ray-project/ray/commit/2dc3cd5b5c2a1719dab96b092aa6737ec74b2b78

also why it is not tested anywhere? how do we know if the mem profiling feature still works today?

Reply:
we can probably just remove the other memray line too.. might even resolve (or mitigate) the py3.11 dependency issue..

Reply:
it's tested on x86_64, the line i removed is for aarch64; we don't do any testing on aarch64 for the most parts, just wheel, docker builds, and post-wheel tests

Reply:
> we don't do any testing on aarch64 for the most parts,

but it means that on aarch64, memory profiling related code won't work?

I don't think `memray` is only used in test only code. it is more like a "soft" dependency of some part of the code, but I am not sure how important it is to omit. if the dependency cannot be resolved, it means that those code parts might fail to work in some circumstances.

and on the other hand, the reason it starts failing, is probably that it is not pinned to a particular commit, but reading from the git repo's default branch's HEAD.

Reply:
> but it means that on aarch64, memory profiling related code won't work?

It might or might not work ;), even if I keep memray and resolve the dependencies for aarch64, we still never know if it ever works because it is not used in any ci build or test jobs for aarch64. It is also not a dependency of the wheel or the docker images, so to use it users need to install memray themselves.

For why we do not have an memray-related tests in CI for aarch64 I'm not sure. Maybe we can only guarantee the core part of ray works in aarch64.

The master version of memray requires a newer versions of ale-py and importlib-metadata that we are pinning to a 1 year old versions.

Reply:
> it seems that serve has a "memory profiling" feature, where it tries to import memray

That's true, but it's optional. We provide it as a debugging convenience and [ask users to `pip install` it](https://github.com/ray-project/ray/blob/6b9435a293fcdc8bf9cde74ac984e870210d290e/python/ray/serve/_private/logging_utils.py#L317-L321) if they need it.

There's no straightforward way to unit test it automatically, as far as I know. When it's used as a debugging tool, we use the CLI to export memory graphs and visually inspect them.

Reply:
@shrekris-anyscale the question is more like do we expect it to work for users on arm machines? how critical is that user journey?

there is a dependency conflict right now, so users likely will get into dependency resolve hell if they try to pip install the package.

Reply:
It's not critical to the user journey. I think it's okay if it doesn't work that well on arm machines.